lib/components/MiningDashboard.js

```diff
@@ -70,19 +66,23 @@ class MiningDashboard extends React.PureComponent {
 
   async mineBlocks(e) {
     e.preventDefault();
-    const { savedCoinbase, numBlocks } = this.state;
+    const { numBlocks } = this.state;
+    const { coinbaseAddress } = this.props;
     try {
-      if (!savedCoinbase) {
+      if (!coinbaseAddress) {
         alert('Must set a coinbase address to mine');
       } else {
         this.setState({ mining: true });
-        this.client
-          .execute('generatetoaddress', [parseInt(numBlocks), savedCoinbase])
-          .then(async result => {
-            const mining = await this.client.execute('getgenerate');
-            this.setState({ mining, numBlocks: 0 });
-          })
-          .catch(e => this.setState({ mining: false }));
+        try {
+          await this.client.execute('generatetoaddress', [
+            parseInt(numBlocks),
```

Comment:
a bit of a nit, but a linter that I've used enforced the radix argument, like `parseInt(numBlocks, 10)`. it has no effect in this case and just adds explicit information